skyportal/handlers/api/spectrum.py

```diff
@@ -517,14 +526,52 @@ def get(self, obj_id):
             .filter(Spectrum.obj_id == obj_id)
             .all()
         )
+
         return_values = []
         for spec in spectra:
             spec_dict = recursive_to_dict(spec)
+            comments = (
+                CommentOnSpectrum.query_records_accessible_by(
+                    self.current_user,
+                    options=[joinedload(CommentOnSpectrum.groups)],
+                )
+                .filter(CommentOnSpectrum.spectrum_id == spec.id)
+                .all()
+            )
+            for c in comments:
+                author = (
+                    User.query_records_accessible_by(
+                        self.current_user,
+                    )
+                    .filter(User.id == c.author_id)
+                    .first()
+                )
+                c.author = author
```

Comment:
```suggestion
```
you should simply be able to access the `author` relationship below on any comment record